drake/systems/plants/RigidBodySystem.cpp

```diff
@@ -463,41 +496,59 @@ RigidBodyDepthSensor::RigidBodyDepthSensor(
     parseScalarValue(range_node, "max", max_range);
   }
 
+  cacheRaycastEndpoints();
+}
+
+void RigidBodyDepthSensor::cacheRaycastEndpoints() {
   raycast_endpoints.resize(3, num_pixel_rows * num_pixel_cols);
   for (size_t i = 0; i < num_pixel_rows; i++) {
     double pitch =
         min_pitch +
         (num_pixel_rows > 1 ? static_cast<double>(i) / (num_pixel_rows - 1)
                             : 0.0) *
-            (max_pitch - min_pitch);
+        (max_pitch - min_pitch);
     for (size_t j = 0; j < num_pixel_cols; j++) {
       double yaw =
           min_yaw +
           (num_pixel_cols > 1 ? static_cast<double>(j) / (num_pixel_cols - 1)
                               : 0.0) *
-              (max_yaw - min_yaw);
+          (max_yaw - min_yaw);
       raycast_endpoints.col(num_pixel_cols * i + j) =
           max_range *
           Vector3d(
               cos(yaw) * cos(pitch), sin(yaw),
               -cos(yaw) *
-                  sin(pitch));  // rolled out from roty(pitch)*rotz(yaw)*[1;0;0]
+              sin(pitch));  // rolled out from roty(pitch)*rotz(yaw)*[1;0;0]
     }
   }
 }
 
+
+
 Eigen::VectorXd RigidBodyDepthSensor::output(const double &t,
                                              const KinematicsCache<double> &rigid_body_state,
                                              const RigidBodySystem::InputVector<double>& u) const {
-  VectorXd distances(num_pixel_cols*num_pixel_rows);
+  const size_t num_distances = num_pixel_cols * num_pixel_rows;
+  VectorXd distances(num_distances);
+
   Vector3d origin = sys.getRigidBodyTree()->transformPoints(rigid_body_state,
                                                              Vector3d::Zero(),
-                                                             frame->frame_index, 0);
+                                                             frame->frame_index,0);
+
   auto raycast_endpoints_world = sys.getRigidBodyTree()->transformPoints(rigid_body_state,
                                                                           raycast_endpoints,
-                                                                          frame->frame_index, 0);
-  sys.getRigidBodyTree()->collisionRaycast(rigid_body_state, origin, raycast_endpoints, distances);
-  // maybe: could publish lcm here (for debugging), since I can easily infer the 3D location in space
+                                                                          frame->frame_index,0);
+
+  sys.getRigidBodyTree()->collisionRaycast(rigid_body_state, origin, raycast_endpoints_world, distances);
```

Comment:
@psiorx added a lidar_returns field to lcmt_quadrotor_output_t which is published on the channel QUAD_OUTPUT.

I added a custom Python snippet to:

1) subscribe to QUAD_OUTPUT
2) given a hard coded theta range from -3pi/4 to 3pi/4, project the lidar_returns with cos, sin
3) given a hard coded name of the sensor frame, transform the projected points to world
4) draw lidar points

To re-use this for other programs, the drawing lcm messages will have to be extended to include the above information that I hard coded.  The theta start, theta step, and sensor frame name.  Note, in this case, the sensor frame is not actually published by the draw commands, only the quad base link is, so I also had to hard code the sensor frame to base link frame.

Maybe @psiorx could post the Python code someplace, I wrote it on his computer.